hubstorage/resourcetype.py

```diff
@@ -40,7 +44,7 @@ def apiget(self, _path=None, **kwargs):
         return self.apirequest(_path, method='GET', **kwargs)
```

Comment:
> I disagree, we should set the GET default as late as possible. Direct calls to _iter_lines also benefits from the retrying without the user duplicating the setdefault code. (in this case iter_json).
> 
> I fixed the apidelete() with kwargs.setdefault, thanks!

Interesting thinking, you accept to set idempotent on apidelete() for DELETE method, but doesn't on apiget() for GET.

For me the "late as possible" is apirequest(), iter_lines() should transparently forward the idempotent kwarg to self.client.request() (a single method, not two, using same api dialect than others: a keyword argument) and callers of _iter_lines() should choose to set idempotent if needed. 

`iter_json` is that case and it has its own retry logic that may not play nice with global retrier, so for a first iteration I think it is better to leave it out of this PR.